packages/@aws-cdk/toolkit-lib/lib/toolkit/toolkit.ts

```diff
@@ -1281,6 +1283,43 @@ export class Toolkit extends CloudAssemblySourceBuilder {
     }
   }
 
+  /**
+   * Retrieve feature flag information from the cloud assembly
+   */
+  public async flags(cx: ICloudAssemblySource): Promise<FeatureFlagReportProperties[]> {
+    const ioHelper = asIoHelper(this.ioHost, 'flags');
+    const assembly = await assemblyFromSource(ioHelper, cx);
+    const artifacts = assembly.cloudAssembly.manifest.artifacts;
+    const allFeatureFlags: ArtifactManifest[] = [];
+
+    for (const [_, artifactInfo] of Object.entries(artifacts!)) {
+      if (artifactInfo.type === ArtifactType.FEATURE_FLAG_REPORT) {
+        allFeatureFlags.push(artifactInfo);
+      }
+    }
+
+    const featureFlagReport: FeatureFlagReportProperties[] = [];
+
+    for (const report of allFeatureFlags) {
+      const flagData = report.properties as any;
```

Comment:
Don't do `as any`. Type errors hide every time you use `as any`. Cast it to the `cxschema.FeatureFlagReportProperties` object you know that it is.